src/orient/rational-calc.orient

```diff
@@ -0,0 +1,273 @@
+
+  rig_storage_min = rig_storage_min_tb * 1024 * 1024 * 1024 * 1024
+  rig_sectors_min = rig_storage_min / sector_size
+  rig_storage_parallelization_min = post_leaves_access / polling_time
+
+  rig_storage_unit_min = rig_storage_min / rig_storage_parallelization_min
+  rig_storage_unit_min_tb = rig_storage_unit_min / (1024 * 1024 * 1024 * 1024)
+
+Crypto:
+  describe(snark_size, "Size of a SNARK proof (bytes)")
+
+Utils:
+  describe(cid_size, "Size of a CID (bytes)")
+
+  b_in_gb = (1024 * 1024 * 1024)
+  secs_in_month = (24 * 30 * 60 * 60)
+  cid_size = 32 + 1 + 1 + 1
+
+ProofOfSpacetime:
+  // post_vanilla_size = node_size * (log2(nodes) + 1) * post_challenges
+  Security:
+    describe(post_challenges, "Number of challenges in a PoRep proof")
+
+  Time:
+    describe(post_time, "Time to generate a PoSt Proof on a single core CPU")
+    describe(post_time_parallel, "Parallel time to generate a PoSt proof on RIG_CORES CPU cores")
+
+    post_time = post_leaves_access + post_proof_gen
+    post_time_parallel = post_leaves_access_parallel + post_proof_gen_parallel
+
+    DiskLatency:
+      describe(post_leaves_access, "Time to retrieve all challenged nodes during PoSt")
+      describe(post_leaves_access_parallel, "Parallel time to retrieve all challenged nodes during PoSt")
+
+      post_leaves_access = post_challenges * rig_storage_latency
+      post_leaves_access_parallel = post_leaves_access / rig_storage_parallelization
+
+    // TODO: add MerkleTreeCaching for ColumnR
+    MerkleTreeCaching:
+      Storage:
+        describe(post_mtree_cached, "Size of the cached parts of the replica merkle tree (bytes)")
+        describe(post_mtree_cached_hashnodes, "Number of intermediary hash nodes cached of the replica merkle tree")
+        describe(post_mtree_cached_gb, "POST_MTREE_CACHED in GiB")
+        describe(post_mtree_overhead, "Fraction of extra storage a miner will have to use to cache the replica merkle tree")
+        describe(post_mtree_layers_cached, "Number of layers of the replica merkle tree being cached")
+        describe(post_mtree_layers_deleted, "Number of layers of the replica merkle tree not being cached")
+
+        post_mtree_cached = post_mtree_cached_hashnodes * node_size
+        post_mtree_cached_gb = post_mtree_cached / (1024 * 1024 * 1024)
+        post_mtree_overhead = post_mtree_cached / sector_size
+        post_mtree_cached_hashnodes = expt(2, post_mtree_layers_cached) - 1
+        post_mtree_layers_deleted = commr_tree_depth - post_mtree_layers_cached
+
+      Time:
+        // Note: this assumes reading the cached parts of the merkle tree instantaneous
+        describe(post_witness_gen, "Time to regenerate and to read the merkle tree inclusion proofs for the challenged nodes from memory")
+
+        post_witness_gen = post_mtree_regen_challenges_time + post_mtree_access_challenges_time
```

Comment:
This accounts for the time required to regenerate a partial merkle tree, but it doesn't account for the time required to regenerate the leaf in the wrapper layer.

Reply:
It does not, this assumed, you already have the wrapper layer. I will be adding the rest soon.

Reply:
added in latest